x/gamm/pool-models/balancer/amm.go

```diff
@@ -328,6 +328,53 @@ func (p *Pool) CalcJoinPoolShares(_ sdk.Context, tokensIn sdk.Coins, swapFee sdk
 	return numShares, newLiquidity, nil
 }
 
+// JoinPoolNoSwap TODO: JoinPoolNoSwap
+func (p *Pool) JoinPoolNoSwap(_ctx sdk.Context, tokensIn sdk.Coins, swapFee sdk.Dec) (numShares sdk.Int, err error) {
+	numShares, newLiquidity, err := p.CalcJoinPoolSharesNoSwap(_ctx, tokensIn, swapFee)
+	if err != nil {
+		return sdk.Int{}, err
+	}
+
+	// update pool with the calculated share and liquidity needed to join pool
+	p.IncreaseLiquidity(numShares, newLiquidity)
+	return numShares, nil
+}
+
+// CalcJoinPoolSharesNoSwap TODO: JoinPoolNoSwap
+func (p *Pool) CalcJoinPoolSharesNoSwap(_ sdk.Context, tokensIn sdk.Coins, swapFee sdk.Dec) (numShares sdk.Int, newLiquidity sdk.Coins, err error) {
```

Comment:
Right now, this is just a copy of `CalcJoinPoolShares` without the logic to single asset join the remaining coins.

Should refactor to reduce code duplication in a separate PR